textvalue/src/main/kotlin/TextValue.kt

```diff
@@ -0,0 +1,86 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
+package com.redmadrobot.textvalue
+
+import android.content.Context
+import android.content.res.Resources
+import android.os.Parcelable
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.compose.runtime.Immutable
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Wrapper to make it possible to work with plain [String] and [StringRes] in the same way.
+ *
+ * ```
+ *  // in some place where we can't access Context
+ *  val errorMessage = TextValue(exception.message, defaultResourceId= R.string.unknown_error)
+ *  showMessage(errorMessage)
+ *
+ *  // in Activity, Fragment or View
+ *  val messageText = getString(message)
+ * ```
+ */
+@Immutable
+public sealed interface TextValue : Parcelable {
+
+    /** Retrieves [String] using the given [context]. */
+    @Deprecated("Use get with Resources instead.", ReplaceWith("this.get(context.resources)"))
+    public fun get(context: Context): String = get(context.resources)
+
+    /** Retrieves [String] using the given [resources]. */
+    public fun get(resources: Resources): String
+
+    override fun equals(other: Any?): Boolean
+    override fun hashCode(): Int
+
+    /** Plain string. */
+    @Parcelize
+    @JvmInline
+    public value class Plain(public val string: String) : TextValue {
```

Comment:
I can't remember why I've made these classes inline 🤔

Reply:
added this with the thought: so far the ideal candidate for value class and therefore inline. Then when we will add args for formatting and plurals subclass - most likely value class will be changed to data class and inline will also go away.

Reply:
removed